src/ui/components/SecondaryToolbox/ReduxDevTools.tsx

```diff
@@ -133,6 +138,96 @@ function isFrameInDecl(decl: ApplyMiddlewareDecl, frame: PauseFrame) {
   );
 }
 
+function isNestedInside(child: SourceLocationRange, parent: SourceLocationRange) {
+  const startsBefore =
+    parent.begin.line < child.begin.line ||
+    (parent.begin.line === child.begin.line && parent.begin.column <= child.begin.column);
+  const endsAfter =
+    parent.end.line > child.end.line ||
+    (parent.end.line === child.end.line && parent.end.column >= child.end.column);
+
+  return startsBefore && endsAfter;
+}
+
+async function searchingCallstackForDispatch(
+  pauseFrames: PauseFrame[],
+  replayClient: ReplayClientInterface,
+  sourcesState: SourcesState
+) {
+  // The first 2 elements in filtered pause frames are from replay's redux stub, so they should be ignored
+  // The 3rd element is the user function that calls it, and will most likely be the `dispatch` call
+  let preferredFrameIdx = 2;
+  const applyMiddlwareDecl = await applyMiddlwareDeclCache.readAsync(replayClient, sourcesState);
+
+  for (let frameIdx = 2; frameIdx < pauseFrames.length; frameIdx++) {
+    const frame = pauseFrames[frameIdx];
+
+    if (isFrameInDecl(applyMiddlwareDecl, frame)) {
+      // this is the frame inside `applyMiddleware` where the initial dispatch occurs
+      // the frame just before this one is the user `dispatch`
+      preferredFrameIdx = frameIdx + 1;
+      return preferredFrameIdx;
+    }
+  }
+
+  return null;
+}
+
+async function isReduxMiddleware(sourceContents: string, location: Location) {
+  const sourceOutline = await sourceOutlineCache.readAsync(replayClient, location.sourceId);
+  const dispatchFn = findFunctionOutlineForLocation(location, sourceOutline);
+  const functions = sourceOutline.functions;
+
+  if (dispatchFn) {
+    const index = functions.indexOf(dispatchFn);
+    if (index >= 2) {
```

Comment:
:question: Let me see if I understand the intent here:

- We're looking for a triply-nested function declaration
- We should be paused somewhere inside the center one
- That implies there are at least two function declarations in the file prior to this one
- So, the source outline's `functions` array _ought_ to be listed sequentially, and thus there's no point in looking if this function isn't at least at index 2

That cover it?

Reply:
> That implies there are at least two function declarations in the file prior to this one

Not sure if I understand how this would always be true. Without this condition wouldn't we need to ensure that `wrapDispatchFn` and `middlewareFn` are not undefined?

Reply:
I'm not saying to change anything here, I'm just trying to make sure I understand the intent of _why_ the `index >= 2` check exists

Reply:
Sorry, yep that's the intent of the check